pkg/test/deployment/yaml.go

```diff
@@ -22,19 +22,26 @@ import (
 )
 
 // NewYamlDeployment creates a new yaml-based deployment.
-func NewYamlDeployment(a *kube.Accessor, kubeConfig, namespace, yamlFile string) (*Instance, error) {
+func NewYamlDeployment(a *kube.Accessor, kubeConfig, namespace, yamlFile string) (out *Instance, err error) {
 	instance := &Instance{}
 
 	instance.kubeConfig = kubeConfig
 	instance.namespace = namespace
 	instance.yamlFilePath = yamlFile
 
+	// If anything goes wrong, undeploy the resources.
+	defer func() {
+		if err != nil {
+			_ = instance.Delete(a, true)
```

Comment:
Nit: it maybe better to let the framework handle the deletion. If no-cleanup is specified, this may end up deleting state that maybe important for diagnostic purposes.

Reply:
It is probably better to have an explicit Deploy operation then (separate from creation). The framework can skip deletion if Deploy returns an error, and any other caller can call Close/Delete themselves explicitly.